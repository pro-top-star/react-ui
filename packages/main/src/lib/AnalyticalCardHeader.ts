import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
