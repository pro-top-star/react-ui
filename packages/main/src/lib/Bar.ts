import { Bar } from '../components/Bar';

export { Bar };
