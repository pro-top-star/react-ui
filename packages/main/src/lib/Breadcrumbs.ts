import { Breadcrumbs } from '../components/Breadcrumbs';

export { Breadcrumbs };
