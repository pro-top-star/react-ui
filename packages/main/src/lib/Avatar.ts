import { Avatar } from '../webComponents/Avatar';

export { Avatar };
