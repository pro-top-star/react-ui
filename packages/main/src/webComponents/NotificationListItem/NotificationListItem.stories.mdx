import { Title, Subtitle, Description, Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import { Fragment } from 'react';
import { NotificationListItem } from '@ui5/webcomponents-react/lib/NotificationListItem';
import { Avatar } from '@ui5/webcomponents-react/lib/Avatar';
import { AvatarSize } from '@ui5/webcomponents-react/lib/AvatarSize';
import { Priority } from '@ui5/webcomponents-react/lib/Priority';
import { List } from '@ui5/webcomponents-react/lib/List';
import { NotificationOverflowAction } from '@ui5/webcomponents-react/lib/NotificationOverflowAction';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { DocsHeader } from '@shared/stories/DocsHeader';

<Meta
  title="UI5 Web Components / NotificationListItem"
  component={NotificationListItem}
  subcomponents={{ NotificationOverflowAction }}
  argTypes={{
    ...createSelectArgTypes({ priority: Priority }),
    avatar: { type: null },
    footnotes: { type: null },
    actions: { type: null }
  }}
  args={{
    priority: Priority.None,
    heading:
      'New order (#2525) With a very long title - Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent feugiat, turpis vel scelerisque pharetra, tellus odio vehicula dolor, nec elementum lectus turpis at nunc.',
    children:
      'And with a very long description and long labels of the action buttons - Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent feugiat, turpis vel scelerisque pharetra, tellus odio vehicula dolor, nec elementum lectus turpis at nunc.'
  }}
/>

<DocsHeader />

<Canvas>
  <Story name="Default">
    {(args) => (
      <List>
        <NotificationListItem
          {...args}
          avatar={
            <Avatar
              size={AvatarSize.XS}
              image="https://sap.github.io/ui5-webcomponents/assets/images/avatars/woman_avatar_1.png"
            />
          }
          footnotes={
            <Fragment>
              <span>Monique Legrand</span>
              <span>2 Days</span>
            </Fragment>
          }
        />
        <NotificationListItem
          showClose
          heading="New order (#2526) With a very long title - Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent feugiat, turpis vel scelerisque pharetra, tellus odio vehicula dolor, nec elementum lectus turpis at nunc."
          priority={Priority.High}
          avatar={
            <Avatar
              size={AvatarSize.XS}
              image="https://sap.github.io/ui5-webcomponents/assets/images/avatars/man_avatar_1.png"
            />
          }
          footnotes={
            <Fragment>
              <span>Alain Devalier</span>
              <span>2 Days</span>
            </Fragment>
          }
        >
          And with a very long description and long labels of the action buttons - Lorem ipsum dolor sit amet,
          consectetur adipiscing elit. Praesent feugiat, turpis vel scelerisque pharetra, tellus odio vehicula dolor,
          nec elementum lectus turpis at nunc.
        </NotificationListItem>
      </List>
    )}
  </Story>
</Canvas>

<ArgsTable story="." />
