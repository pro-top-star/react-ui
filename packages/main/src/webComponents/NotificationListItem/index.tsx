import { Priority } from '@ui5/webcomponents-react/lib/Priority';
import { withWebComponent, WithWebComponentPropTypes } from '@ui5/webcomponents-react/lib/withWebComponent';
import '@ui5/webcomponents-fiori/dist/NotificationListItem';
import { FC, ReactNode } from 'react';

export interface NotificationListItemPropTypes extends WithWebComponentPropTypes {
  /**
   * Defines if the `notification` is new or has been already read.
   *
   * **Note:** if set to `false` the `heading` has bold font, if set to true - it has a normal font.
   */
  read?: boolean;
  /**
   * Defines if the `heading` and `description` should wrap, they truncate by default.
   *
   * **Note:** by default the `heading` and `decription`, and a `ShowMore/Less` button would be displayed.
   */
  wrap?: boolean;
  /**
   * Defines if a busy indicator would be displayed over the item.
   */
  busy?: boolean;
  /**
   * Defines the `heading` of the item.
   */
  heading?: string;
  /**
   * Defines the `priority` of the item.
   */
  priority?: Priority;
  /**
   * Defines if the `close` button would be displayed.
   */
  showClose?: boolean;
  /**
   * Defines the selected state of the `ListItem`.
   */
  selected?: boolean;
  /**
   * Defines the avatar, displayed in the `NotificationListItem`.
   *
   * **Note:** Consider using the `Avatar` to display icons, initials or images.
   */
  avatar?: ReactNode;
  /**
   * Defines the content of the `NotificationListItem`, usually a description of the notification.
   *
   * **Note:** Аlthough this slot accepts HTML Elements, it is strongly recommended that you only use text in order to preserve the intended design.
   */
  children?: ReactNode | ReactNode[];
  /**
   * Defines the elements, dipalyed in the footer of the of the `NotificationListItem`.
   */
  footnotes?: ReactNode | ReactNode[];
  /**
   * Defines the actions, displayed in the top-right area.
   *
   * **Note:** use the `NotificationAction` component.
   */
  actions?: ReactNode;
  /**
   * Fired when the `Close` button is pressed.
   */
  onClose?: (event: CustomEvent) => void;
}

/**
 * The `NotificationListItem` is a type of list item, meant to display notifications.
 * The component has a rich set of various properties that allows the user to set `avatar`, `heading`, descriptive `content` and `footnotes` to fully describe a notification.
 * The user can:
 *
 * *   display a `Close` button
 * *   can control whether the `heading` and `description` should wrap or truncate and display a `ShowMore` button to switch between less and more information
 * *   add custom actions by using the `NotificationAction` component
 *
 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/NotificationListItem" target="_blank">UI5 Web Components Playground</a>
 */
const NotificationListItem: FC<NotificationListItemPropTypes> = withWebComponent<NotificationListItemPropTypes>(
  'ui5-li-notification',
  ['heading', 'priority'],
  ['read', 'wrap', 'busy', 'showClose', 'selected'],
  ['avatar', 'footnotes', 'actions'],
  ['close']
);

NotificationListItem.displayName = 'NotificationListItem';

NotificationListItem.defaultProps = {
  read: false,
  wrap: false,
  busy: false,
  priority: Priority.None,
  showClose: false,
  selected: false
};

export { NotificationListItem };
