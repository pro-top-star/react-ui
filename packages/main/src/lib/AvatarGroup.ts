import { AvatarGroup } from '../webComponents/AvatarGroup';
import type { AvatarGroupPropTypes } from '../webComponents/AvatarGroup';

export { AvatarGroup };
export type { AvatarGroupPropTypes };
