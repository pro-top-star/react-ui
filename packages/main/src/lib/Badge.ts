import { Badge } from '../webComponents/Badge';
import type { BadgePropTypes } from '../webComponents/Badge';

export { Badge };
export type { BadgePropTypes };
