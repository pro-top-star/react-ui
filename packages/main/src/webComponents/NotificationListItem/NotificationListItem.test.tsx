import { NotificationListItem } from '@ui5/webcomponents-react/lib/NotificationListItem';
import { mount } from 'enzyme';
import React from 'react';

describe('NotificationListItem', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mount(<NotificationListItem />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
