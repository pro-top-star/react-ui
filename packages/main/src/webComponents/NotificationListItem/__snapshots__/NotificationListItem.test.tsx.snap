// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationListItem Basic Test (generated) 1`] = `
<DocumentFragment>
  <ui5-li-notification
    heading="My Notification"
    priority="High"
    show-close="true"
  >
    <div
      slot="actions"
      style="display: contents;"
    >
      <ui5-notification-action
        design="Transparent"
        text="My Action"
        ui5-notification-action=""
      />
    </div>
  </ui5-li-notification>
</DocumentFragment>
`;
